packages/material-ui/src/MenuList/MenuList.js

```diff
@@ -6,66 +6,95 @@ import ReactDOM from 'react-dom';
 import warning from 'warning';
 import ownerDocument from '../utils/ownerDocument';
 import List from '../List';
+import getScrollbarSize from 'dom-helpers/util/scrollbarSize';
+
+function MenuList({ actions, children, className, onBlur, onKeyDown, disableListWrap, ...other }) {
+  const [currentTabIndex, setCurrentTabIndex] = React.useState(null);
+  const blurTimerRef = React.useRef();
+  const listRef = React.useRef();
+  const selectedItemRef = React.useRef();
+
+  React.useImperativeHandle(actions, () => ({
+    focus: () => {
+      if (selectedItemRef.current) {
+        ReactDOM.findDOMNode(selectedItemRef.current).focus();
+        return;
+      }
 
-class MenuList extends React.Component {
-  state = {
-    currentTabIndex: null,
-  };
-
-  componentDidMount() {
-    this.resetTabIndex();
-  }
-
-  componentWillUnmount() {
-    clearTimeout(this.blurTimer);
-  }
-
-  setTabIndex(index) {
-    this.setState({ currentTabIndex: index });
-  }
-
-  handleBlur = event => {
-    this.blurTimer = setTimeout(() => {
-      if (this.listRef) {
-        const list = this.listRef;
+      if (listRef.current && listRef.current.firstChild) {
+        listRef.current.firstChild.focus();
+      }
+    },
+    getContentAnchorEl: () => {
```

Comment:
What's the use case for this and `adjustStyleForScrollbar`. Didn't we only discuss the `focus` handle?

Reply:
They would potentially be used in the [MenuList composition](https://material-ui.com/demos/menus/#menulist-composition) use case (e.g. Popper-based menu), but the primary reason for `getContentAnchorEl` and `adjustStyleForScrollbar` was to decouple `Menu` from the `MenuList` implementation details and get rid of `Menu` having references to `MenuList` instance properties. Moving the logic for `adjustStyleForScrollbar` into `MenuList` also made it much easier to test and allowed me to remove somewhat convoluted spying/stubbing approaches from `Menu.test.js`.

Reply:
Makes sense. I guess the important part is https://github.com/mui-org/material-ui/pull/14865/files#diff-649b5b30137fa3f72934fc6d8077c444R59.

Maybe this is just me but it helps if these things have different commits in a PR. The commits themself can break the build e.g. if you simply refactor and tests break then you can fix the tests in a different commit if the fix would only be temporary. It is just much easier to review this if I know what change belongs together.

Reply:
I'll keep that in mind for the future and do smaller commits that line up closer with the separate steps I took in doing the work.